csrc/fusion_segmenter.cpp

```diff
@@ -3548,15 +3549,28 @@ bool SegmentCandidateFinder::codeGenSupportedMerge(
   NVF_ERROR(
       areDirectlyConnected(group1, group2),
       "only support testing immediate producer-consumer groups");
-  auto h = tryMerge(segmented_fusion_.get(), runtime_info_, group1, group2);
+  if (options_.only_segment_resharding_exprs) {
```

Comment:
In addition to merging, I'm worried about [this code path](https://github.com/NVIDIA/Fuser/blob/1603f39bab8c1bbe12e38f2b5de53dec3b7cc373/csrc/fusion_segmenter.cpp#L1873) where the segmenter first tries to schedule the entire fusion, unsegmented. 

Did you test this? For example, create a simple pointwise fusion that doesn't require segmentation, and change some tensors to distributed. Code as is might end up segmenting the whole fusion as one kernel.

Reply:
It seems that this path doesn't expose the`SegmentCandidateFinderOptions options`, so technically this path is not reachable in our case, and so the segmenter should behave as usual.

This being said, in a more general way, we always hit the problem that manual multi-device scheduling cannot be composed with the scheduler.

Reply:
> technically this path is not reachable in our case, and so the segmenter should behave as usual.

Yes and yes, and that's exactly the problem. I think code as is will **miss** segmentation that's **necessary** for multi-device. Am I understanding it correctly?

Reply:
> the multidevice runtime will always call the segmenter in a proper way

Thanks -- that's what I missed. To make sure I understand you, you are saying the multidevice runtime never calls [this flavor](https://github.com/NVIDIA/Fuser/blob/1603f39bab8c1bbe12e38f2b5de53dec3b7cc373/csrc/fusion_segmenter.h#L581-L584) of `SegmentCandidateFinder::segment`, the only flavor that tries to schedule the whole, unsegmented fusion. Am I understanding it correctly?

Reply:
Yes, exactly. The only way the multi-device segmenter is called is as shown in the test [here](https://github.com/NVIDIA/Fuser/blob/ddb7a193ec603869d5ae1695a47bdb839abfc32d/test/test_pipeline.cpp#L410)